plugins/modules/ntnx_acps.py

```diff
@@ -0,0 +1,449 @@
+                    operator: IN,
+                    right_hand_side:
+                        uuid_list:
+                            00000000000-0000-0000-0000-00000000000
+        role_reference:
+            kind: role,
+            uuid: 00000000000-0000-0000-0000-00000000000
+        user_group_reference_list:
+            kind: user_group,
+            uuid: 00000000000-0000-0000-0000-00000000000
+        user_reference_list:
+                kind: user,
+                uuid: 00000000000-0000-0000-0000-00000000000
+status:
+  description: An intentful representation of a ACP status
+  returned: always
+  type: dict
+  sample:
+    description: desc
+    execution_context:
+        task_uuid:
+            00000000000-0000-0000-0000-00000000000
+    is_system_defined: false
+    name: name
+    resources:
+        filter_list:
+            context_list:
+                entity_filter_expression_list:
+                    left_hand_side:
+                        entity_type: ALL
+                    operator: IN
+                    right_hand_side:
+                        collection: ALL
+                scope_filter_expression_list:
+                    left_hand_side: PROJECT,
+                    operator: IN,
+                    right_hand_side:
+                        uuid_list:
+                            00000000000-0000-0000-0000-00000000000
+        role_reference:
+            kind: role,
+            name: Project Admin,
+            uuid: 00000000000-0000-0000-0000-00000000000
+        user_group_reference_list:
+                kind: user_group,
+                name: cn=sspadmins,cn=users,dc=qa,dc=nucalm,dc=io,
+                uuid: 00000000000-0000-0000-0000-00000000000
+        user_reference_list:
+                kind: user,
+                name: idpuser1@calmsaastest.com,
+                uuid: 00000000000-0000-0000-0000-00000000000
+    state: COMPLETE
+acp_uuid:
+  description: The created security rule  uuid
+  returned: always
+  type: str
+  sample: 00000000000-0000-0000-0000-00000000000
+task_uuid:
+  description: The task uuid for the creation
+  returned: always
+  type: str
+  sample: 00000000000-0000-0000-0000-00000000000
+"""
+
+from ..module_utils import utils  # noqa: E402
+from ..module_utils.base_module import BaseModule  # noqa: E402
+from ..module_utils.prism.tasks import Task  # noqa: E402
+from ..module_utils.prism.acps import ACP  # noqa: E402
+from ..module_utils.utils import remove_param_with_none_value  # noqa: E402
+
+
+def get_module_spec():
+
+    mutually_exclusive = [("name", "uuid")]
+
+    entity_by_spec = dict(name=dict(type="str"), uuid=dict(type="str"))
+
+    rhs_spec = dict(
+        collection=dict(type="str", choices=["ALL", "SELF_OWNED"]),
+        categories=dict(type="dict"),
+        uuid_list=dict(type="list"),
+    )
+
+    scope_context_spec = dict(
+        lhs=dict(type="str", choices=["CATEGORY", "PROJECT", "CLUSTER", "VPC"]),
```

Comment:
We have more options I believe. Better to remove choices. In examples mention some which we were able to identify